java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/MainActivity.java

```diff
@@ -0,0 +1,111 @@
+package com.example.chartboost.mediation.sdk.demo.java;
+
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.activity.EdgeToEdge;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.chartboost.mediation.sdk.demo.java.databinding.ActivityMainBinding;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity implements UILogsListener, OnInitializationCompleteListener {
+
+    private ActivityMainBinding binding;
+    private final AdController adController = new AdController();
+    private RecyclerView logRecyclerView;
+    private LogAdapter logAdapter;
+    private final List<String> uiLogs = new ArrayList<>();
+
+    String interstitialPlacementName;
+    String rewardedPlacementName;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        EdgeToEdge.enable(this);
+        setContentView(binding.getRoot());
+        interstitialPlacementName = getString(R.string.interstitial_placement);
+        rewardedPlacementName = getString(R.string.rewarded_placement);
+        setupLogsRecyclerView();
+
+        adController.initializeChartboost(
+                getString(R.string.default_app_id),
+                this,
+                getBaseContext(),
+                this
+        );
+
+        binding.loadInterstitialBtn.setOnClickListener(v -> onLoadClick(interstitialPlacementName, binding.showInterstitialBtn));
+        binding.showInterstitialBtn.setOnClickListener(v -> onShowClick(interstitialPlacementName, (Button) v));
+
+        binding.loadRewardedBtn.setOnClickListener(v -> onLoadClick(rewardedPlacementName, binding.showRewardedBtn));
+        binding.showRewardedBtn.setOnClickListener(v -> onShowClick(rewardedPlacementName, (Button) v));
+
+        binding.queueSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            adController.setShouldUseQueue(isChecked);
+            addLog(String.format("Fullscreen Ad Queue will %s", isChecked ? "be used" : "not be used"));
+        });
+    }
+
+    private void setupLogsRecyclerView() {
+        logRecyclerView = binding.logsRv;
+        Button clearLogsButton = binding.clearLogsBtn;
+        logRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+        logAdapter = new LogAdapter(uiLogs);
+        logRecyclerView.setAdapter(logAdapter);
+        clearLogsButton.setOnClickListener(v -> clearLogs());
+    }
+
+    private void onLoadClick(String placementName, Button correspondingShowButton) {
+        adController.loadFullscreenAd(placementName, getBaseContext(), correspondingShowButton, this);
+    }
```

Comment:
it looks like users can repeatedly click the "Load" button while an ad is already loading. This can lead to multiple load requests firing, which might be confusing or inefficient.